### PR TITLE
Show README and metadata for yanked crates/versions

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves https://github.com/rust-lang/crates.io/issues/6414

### Before

Bildschirmfoto 2023-05-02 um 13 30 16

### After

Bildschirmfoto 2023-05-02 um 13 29 45

I'm opening this as a draft PR for discussion among the team. Also, tests are still missing for this change.

### CODE REVIEW

File: app/components/crate-header.hbs [hunk 1]
```diff
@@ -3,6 +3,13 @@
     <span data-test-crate-name>{{@crate.name}}</span>
     {{#if @version}}
       <small data-test-crate-version>v{{@version.num}}</small>
+
+      {{#if @version.yanked}}
+        <span local-class="yanked-badge">
+          {{svg-jar "trash"}}
+          Yanked
+        </span>
```
Comment (LawnGnome) [inline app/components/crate-header.hbs:11]: Do we want to link to something — or at least have a tooltip — which explains what yanking means in the same way the previous version said "This crate has been yanked, but it is still available for download for other crates that may be depending on it"?

**Review by LawnGnome — APPROVED** [review]
I think this is a good change, provided we take care not to confuse anyone as to what "yanked" actually means. :+1:

File: app/components/crate-header.hbs (hunk 1 above)
Comment (Turbo87) [reply app/components/crate-header.hbs:11]: that sounds like a good idea, thanks! 👍

Comment (Turbo87) [reply app/components/crate-header.hbs]: Bildschirmfoto 2023-05-03 um 16 27 14

**Comment by Turbo87** [comment]
I've modified the existing tests for this route to cover the new behavior.

This PR is ready for review/merging now. Unless there are any objections I plan on merging this some time tomorrow.

**Review by LawnGnome — APPROVED** [review]

**Pull request merged**